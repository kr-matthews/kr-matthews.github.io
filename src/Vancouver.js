function Vancouver(props) {
  return (
    <div className="page narrow-page">
      <h1>Vancouver</h1>
      <p>
        I live in Vancouver, BC, Canada. There are many odd/interesting facts
        about streets, parks, etc. in{" "}
        <a
          href="https://en.wikipedia.org/wiki/Metro_Vancouver_Regional_District"
          target="_blank"
          rel="noopener noreferrer"
        >
          Metro Vancouver
        </a>
        . I've highlight many of them in the map below. Vancouver has also been
        expanding and improving bike infrastructure over the past few decades,
        which I've highlighted too. I'll be adding to this map as I find more
        things.
      </p>
      <h2>Custom Map</h2>
      <iframe
        title="vancouverMap"
        src="https://www.google.com/maps/d/u/0/embed?mid=1VjjBQZ-70CWIjFpoo8BHS_bLgm3xpQQ3"
        width="750"
        height="550"
      ></iframe>
      <h2>Street Name Resources</h2>
      Here are some links with more information:
      {/* // TODO: fix css for list/list items */}
      <ul>
        <li>
          a{" "}
          <a
            href="https://geographic.org/streetview/canada/bc/vancouver.html"
            target="_blank"
            rel="noopener noreferrer"
          >
            list
          </a>{" "}
          of all Vancouver streets
        </li>
        <li>
          an{" "}
          <a
            href="https://newsinteractives.cbc.ca/longform/streets"
            target="_blank"
            rel="noopener noreferrer"
          >
            article
          </a>{" "}
          from the CBC examining the origins of all Vancouver street names
        </li>
      </ul>
    </div>
  );
}

export default Vancouver;
