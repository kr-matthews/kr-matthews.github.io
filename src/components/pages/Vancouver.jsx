import Link from "../common/Link";

export default function Vancouver() {
  return (
    <div className="page narrow-page">
      <h1>Vancouver</h1>
      <p>
        I live in Vancouver, BC, Canada. There are many odd/interesting facts
        about streets, parks, etc. in{" "}
        <Link
          href="https://en.wikipedia.org/wiki/Metro_Vancouver_Regional_District"
          isExternal
        >
          Metro Vancouver
        </Link>
        . I've highlight many of them in the map below. Vancouver has also been
        expanding and improving bike infrastructure over the past few decades,
        which I've highlighted too. I'll be adding to this map as I find more
        things.
      </p>
      <h3>Custom Map</h3>
      <p>
        Click on icons for pictures and information, or open the left panel to
        filter by category.
      </p>
      <iframe
        title="vancouverMap"
        src="https://www.google.com/maps/d/u/0/embed?mid=1VjjBQZ-70CWIjFpoo8BHS_bLgm3xpQQ3"
        width="750"
        height="550"
      ></iframe>
      <h3>Street Name Resources</h3>
      Here are some links with more information:
      <ul>
        <li>
          a{" "}
          <Link
            href="https://geographic.org/streetview/canada/bc/vancouver.html"
            isExternal
          >
            list
          </Link>{" "}
          of all Vancouver streets
        </li>
        <li>
          an{" "}
          <Link
            href="https://newsinteractives.cbc.ca/longform/streets"
            isExternal
          >
            article
          </Link>{" "}
          from the CBC examining the origins of all Vancouver street names
        </li>
      </ul>
    </div>
  );
}
