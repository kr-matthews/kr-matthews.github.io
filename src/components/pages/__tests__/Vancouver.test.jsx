import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Vancouver from "../Vancouver";
import { BrowserRouter } from "react-router-dom";

describe("components/pages/Vancouver", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Vancouver />, {
      wrapper: ({ children }) => <BrowserRouter>{children}</BrowserRouter>,
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
