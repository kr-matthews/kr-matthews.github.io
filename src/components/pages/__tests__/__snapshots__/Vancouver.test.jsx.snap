// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Vancouver matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-rzly3e"
  >
    <div
      class="css-jydvbh"
    >
      <h1>
        Vancouver
      </h1>
      <p
        style="margin-bottom: 2em;"
      >
        I live in Vancouver, BC, Canada.
      </p>
      <div
        class="chakra-tabs css-0"
      >
        <div
          aria-orientation="horizontal"
          class="chakra-tabs__tablist css-k008qs"
          role="tablist"
        >
          <button
            aria-controls="tabs-:r0:--tabpanel-0"
            aria-selected="true"
            class="chakra-tabs__tab css-5wl5ud"
            data-index="0"
            id="tabs-:r0:--tab-0"
            role="tab"
            tabindex="0"
            type="button"
          >
            Bike Route Map
          </button>
          <button
            aria-controls="tabs-:r0:--tabpanel-1"
            aria-selected="false"
            class="chakra-tabs__tab css-5wl5ud"
            data-index="1"
            id="tabs-:r0:--tab-1"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Streets & Facts
          </button>
          <button
            aria-controls="tabs-:r0:--tabpanel-2"
            aria-selected="false"
            class="chakra-tabs__tab css-5wl5ud"
            data-index="2"
            id="tabs-:r0:--tab-2"
            role="tab"
            tabindex="-1"
            type="button"
          >
            GeoGuessr
          </button>
        </div>
        <div
          class="chakra-tabs__tab-panels css-8atqhb"
        >
          <div
            aria-labelledby="tabs-:r0:--tab-0"
            class="chakra-tabs__tab-panel css-sjt5nk"
            id="tabs-:r0:--tabpanel-0"
            role="tabpanel"
            tabindex="0"
          >
            <p>
              One of my 
              <a
                class="chakra-link css-1108m31"
                href="/projects"
              >
                projects
              </a>
              , 
              <a
                class="chakra-link css-1108m31"
                href="https://kr-matthews.github.io/bike-route-map/"
                rel="noopener"
                target="_blank"
                to=""
              >
                Bike Route Map
              </a>
               maps out bike routes/infrastructure in Vancouver and nearby regions. I attach videos to the routes, filmed from a camera on my helmet.
            </p>
            <p>
              I made this because official city resources are often ambiguous or imprecise -- for example, you can't zoom in sufficiently on complicated areas, or distinguish between roads and paths. When combined with the often poor (or completely lacking) signage on routes, it can be very hard to navigate. Google maps street view can help, but it's surprisingly often not clear where you're meant to be (the road? the sidewalk?).
            </p>
          </div>
          <div
            aria-labelledby="tabs-:r0:--tab-1"
            class="chakra-tabs__tab-panel css-sjt5nk"
            hidden=""
            id="tabs-:r0:--tabpanel-1"
            role="tabpanel"
            tabindex="0"
          >
            <p>
              There are many odd/interesting facts about streets, parks, etc. in 
              <a
                class="chakra-link css-1108m31"
                href="https://en.wikipedia.org/wiki/Metro_Vancouver_Regional_District"
                rel="noopener"
                target="_blank"
                to=""
              >
                Metro Vancouver
              </a>
              . I've highlight many of them in the map below. Vancouver has also been expanding and improving bike infrastructure over the past few decades, which I've highlighted too. I'll be adding to this map as I find more things.
            </p>
            <figure
              style="margin-bottom: 1em;"
            >
              <iframe
                height="550"
                src="https://www.google.com/maps/d/u/0/embed?mid=1VjjBQZ-70CWIjFpoo8BHS_bLgm3xpQQ3"
                title="vancouverMap"
                width="750"
              />
              <figcaption
                style="font-size: .8em;"
              >
                Click on icons for pictures and information, or open the left panel to filter by category.
              </figcaption>
            </figure>
            Here are some links with more information:
            <ul
              class="css-tu0njr"
              role="list"
            >
              <li
                class="css-0"
              >
                a 
                <a
                  class="chakra-link css-1108m31"
                  href="https://geographic.org/streetview/canada/bc/vancouver.html"
                  rel="noopener"
                  target="_blank"
                  to=""
                >
                  list
                </a>
                 of all Vancouver streets
              </li>
              <li
                class="css-0"
              >
                an 
                <a
                  class="chakra-link css-1108m31"
                  href="https://newsinteractives.cbc.ca/longform/streets"
                  rel="noopener"
                  target="_blank"
                  to=""
                >
                  article
                </a>
                 from the CBC examining the origins of all Vancouver street names
              </li>
            </ul>
          </div>
          <div
            aria-labelledby="tabs-:r0:--tab-2"
            class="chakra-tabs__tab-panel css-sjt5nk"
            hidden=""
            id="tabs-:r0:--tabpanel-2"
            role="tabpanel"
            tabindex="0"
          >
            <p>
              <a
                class="chakra-link css-1108m31"
                href="https://www.geoguessr.com/maps/6180d71598cffd00011e1e91"
                rel="noopener"
                target="_blank"
                to=""
              >
                GeoGuessr
              </a>
               is an online game which drops you somewhere in google maps street view, and you need to figure out where it is on the map. The world map is the most popular, but you can restrict the range to regions of the world -- I prefer to play 
              <a
                class="chakra-link css-1108m31"
                href="https://www.geoguessr.com/maps/6180d71598cffd00011e1e91"
                rel="noopener"
                target="_blank"
                to=""
              >
                just Vancouver
              </a>
               and 
              <a
                class="chakra-link css-1108m31"
                href="https://www.geoguessr.com/maps/652343aecd435d7e5b5f55e3"
                rel="noopener"
                target="_blank"
                to=""
              >
                Vancouver, Burnaby, & Richmond
              </a>
              . Sometimes larger subsets of Metro Vancouver. I usually do 10s NMPZ (10 seconds per location, with no moving, panning, or zooming -- just a fixed image).
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
